packages/core/src/auth/tokens/token-scope.ts

```diff
@@ -116,3 +117,89 @@ export type TokenScope =
   | typeof AllDataReadWrite
   | Permissions
   | PredefinedScope;
+
+export interface CacheItemKey {
+  key: string;
+}
+
+export interface CacheItemKeyPrefix {
+  keyPrefix: string;
+}
+
+export function isCacheItemKey(
```

Comment:
This check should actually be like below because `key` is a substring of `keyPrefix`
```
return 'key' in cacheItem and !('keyPrefix' in cacheItem);
```
Otherwise, you need to always call `isCacheItemKeyPrefix()` first before `isCacheItemKey()`. And right now in both the nodejs sdk and web sdk, you are calling them in the reverse order. I think this will fix at least some of the failing tests.

Reply:
the `in` here isn't a substring operation though, is it? It is more like a `HashMap.containsKey` thing AFAIK. Will do a quick experiment in the REPL to verify.

Reply:
oh you are right. i was not thinking clearly.